### PR TITLE
Create listpending listcontracts #3 #6 #4

### PR DESCRIPTION
**Pull request by TedGu**
### What this PR does (required):
- CREATE Invitation /user/:id/invitation #3
- GET invitations /user/:id/invitations #6
- GET contacts /user/:id/contacts #4
### Screenshots / Videos (required):
- CREATE Invitation /user/:id/invitation #3 

image

- GET invitations /user/:id/invitations #6

image

- GET contacts /user/:id/contacts #4

image

### Any information needed to test this feature (required):
- CREATE Invitation /user/:id/invitation #3
POST method with toUser, toUserEmail and fromUser(`id` in ` /user/:id/invitation`)

- GET invitations /user/:id/invitations #6
GET method for specific user with user id ( `id` in  `/user/:id/invitations`)

- GET contacts /user/:id/contacts #4
GET method for specific user with user id ( `id` in  `/user/:id/contacts`)

### CODE REVIEW

File: server/routes/userinvitation.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+const router = express.Router();
+
+const { createInvitation, getPendingInvitation, getContracts } = require("../controllers/userinvitation");
+
+router.route("/:id/invitation").post(createInvitation);
+router.route("/:id/invitations").get(getPendingInvitation);
+router.route("/:id/contracts").get(getContracts);
+
+
+module.exports = router;
```
Comment (dproc96) [inline server/routes/userinvitation.js:11]: These and the approve reject routes could probably be in the same router

**Review by dproc96 — COMMENTED** [review]
Had one new comment, my comments from #45 apply here as well though

File: server/models/Invitation.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+  ],
+
+  approved: [
+    {
+      type: Boolean,
+      default: false,
+      required: true,
+    },
+  ],
```
Comment (sina-jamshidi) [inline server/models/Invitation.js:35]: these do not need to be arrays?

**Review by sina-jamshidi — COMMENTED** [review]
Can't approve this until the changes have been made

File: server/controllers/userinvitation.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  } catch (error) {
+    console.log(new Date(), error);
+    res.json({
+      status: 500,
+      message: error,
+    });
+  }
+});
+
+// @route GET /user/:id/contracts
```
Comment (sina-jamshidi) [inline server/controllers/userinvitation.js:56]: typo: `/user/:id/contacts`

File: server/controllers/userinvitation.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+exports.getContacts = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.params.id;
+    const acceptedInvitations = await Invitation.find({
+      toUser: userId,
+      approved: true,
+    });
+    res.json({
+      status: "success",
+      message: "Got accepted invitations successfully",
```
Comment (sina-jamshidi) [inline server/controllers/userinvitation.js:67]: we probably don't need a success message for this

File: server/controllers/userinvitation.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  try {
+    const userId = req.params.id;
+    const pendingInvitations = await Invitation.find({
+      toUser: userId,
+      approved: false,
+      rejected: false,
+    });
+    res.json({
+      status: "success",
+      message: "Got pending invitations successfully",
```
Comment (sina-jamshidi) [inline server/controllers/userinvitation.js:44]: we probably don't need a success message for this

**Review by sina-jamshidi — APPROVED** [review]
This looks good. One small comment but also make sure you merge in your latest model/controller changes from main so that the models aren't arrays again

File: server/routes/userinvitation.js (hunk 1 above)
Comment (TedGu) [reply server/routes/userinvitation.js]: have merged into the same route file.

**Pull request merged**